### PR TITLE
Turn on OSX builds on Travis

### PR DESCRIPTION
**Pull request by rdeits**

### CODE REVIEW

**Comment by rdeits** [comment]
Okay, I believe this is working now. The build works and c++ tests pass with clang on osx. Using gcc on osx caught a bug in which we were using `isinf` instead of `std::isinf`. I've also removed the flag from travis which was allowing gcc to fail at compile time, because it appears to be working again.

**Comment by rdeits** [comment]
Note that I've moved the OS-specific configuration to a new script, `travis_pre_install.sh`. I've also turned off Director in the osx build until https://github.com/RobotLocomotion/director/issues/140 is resolved

File: .travis.yml [hunk 1]
```diff
@@ -34,12 +34,10 @@ python:
...
-    - CMAKE_FLAGS="-DWITH_SPOTLESS:BOOL=OFF -DWITH_LIBBOT:BOOL=ON -DWITH_DIRECTOR:BOOL=ON -DWITH_IRIS:BOOL=ON -DWITH_OCTOMAP:BOOL=ON -DWITH_MOSEK:BOOL=ON -DWITH_AVL:BOOL=ON -DWITH_XFOIL:BOOL=ON"
   matrix:
     - BUILD_TYPE=Release
     # - BUILD_TYPE=Debug
 
 matrix:
-  allow_failures:
-    - compiler: gcc
+  # allow_failures:
+  #   - compiler: gcc
```
Comment (rdeits) [inline .travis.yml]: gcc works again!

File: CMakeLists.txt [hunk 2]
```diff
@@ -113,7 +113,7 @@ set(gurobi_GIT_REPOSITORY https://github.com/RobotLocomotion/gurobi.git)
 set(gurobi_GIT_TAG 9699ec6091f3d9d8013c6f0acaea229c84a189a5)
 set(gurobi_IS_PUBLIC FALSE)
 set(iris_GIT_REPOSITORY https://github.com/rdeits/iris-distro.git)
-set(iris_GIT_TAG a17bdb4a20a8195363b629c47fd1b6251b44c95e)
+set(iris_GIT_TAG 7442fbca7a456f5564296902d8e2130a751bf3e3)
```
Comment (rdeits) [inline CMakeLists.txt]: not building director exposed bugs in iris's `install_prereqs.sh`

File: travis_pre_build.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	sudo apt-get update -qq
+	sudo apt-get install gcc-4.8 g++-4.8
+    export CMAKE_FLAGS="-DWITH_SPOTLESS:BOOL=OFF -DWITH_LIBBOT:BOOL=ON -DWITH_DIRECTOR:BOOL=ON -DWITH_IRIS:BOOL=ON -DWITH_OCTOMAP:BOOL=ON -DWITH_MOSEK:BOOL=ON -DWITH_AVL:BOOL=ON -DWITH_XFOIL:BOOL=ON"
+	mkdir build
+	make download-all
+	sudo ./install_prereqs.sh ubuntu
+elif [ "$TRAVIS_OS_NAME" = "osx" ]
+	then 
+	brew update > brew_update.log
+	export CMAKE_FLAGS="-DWITH_SPOTLESS:BOOL=OFF -DWITH_LIBBOT:BOOL=ON -DWITH_DIRECTOR:BOOL=OFF -DWITH_IRIS:BOOL=ON -DWITH_OCTOMAP:BOOL=ON -DWITH_MOSEK:BOOL=ON -DWITH_AVL:BOOL=ON -DWITH_XFOIL:BOOL=ON"
```
Comment (rdeits) [inline travis_pre_build.sh]: this is the same as Ubuntu, but with Director turned off

File: .travis.yml [hunk 4]
```diff
@@ -1,3 +1,7 @@
+os:
+  - linux
+  - osx
```
Comment (rdeits) [inline .travis.yml]: this is the line that actually creates an osx build

**Comment by rdeits** [comment]
Woohoo! [gcc, clang] x [linux, osx] are all passing on Travis!

**Comment by RussTedrake** [comment]
awesome!  will just wait for the other servers then merge!

**Comment by rdeits** [comment]
Just a timeout on drake004 in Pendulum runLQR

**Pull request merged**